components/properties/src/error.rs

```diff
@@ -28,10 +29,23 @@ pub enum PropertiesError {
     /// An unknown value was used for the [`GeneralCategoryGroup`](crate::GeneralCategoryGroup) property
     #[displaydoc("Unknown general category group: {0}")]
     UnknownGeneralCategoryGroup(u32),
+    /// An unknown or unexpected property name was used for an API dealing with properties specified as strings at runtime
+    #[displaydoc("Unexpected or unknown property name {0}")]
+    UnexpectedPropertyName(TinyStr16),
```

Comment:
Thought/Suggestion: if the input was something like Canonical_Combining_Closs (misspelled), this produces error messages such as

"Unexpected or unknown property name Canonical_Combin"

truncated to 16 characters.

I think I lean toward not including the truncated property name and instead allowing the caller to stick it in an Eyre context or something. We could also `log` it.

```rust
icu_properties::sets::load_for_ecma262(prop_name).map_err(|e| e.with_context(prop_name))?
```